### PR TITLE
Validate support optionals

### PR DESCRIPTION
**Pull request by borft**
This add support for optionals in the validation. Basically, to make a context property optional, add `optional: True`

```
example:
"properties": {
        "referrer": {
          "description": "Full URL to HTTP referrer of the current page.",
          "type": "string",
          "optional": true
        }
}
```

### CODE REVIEW

**Review by thijs-obj — COMMENTED** [review]
Changes look good to me, but do we not want to test the `validate_event_adheres_to_schema()` function too with some optional properties?

**Comment by borft** [comment]
> Changes look good to me, but do we not want to test the `validate_event_adheres_to_schema()` function too with some optional properties?

I think that is a good idea. However, as there are currently no optionals in the schema, I propose to add that as soon as there are (which will be when we add the `MarketingContext`)

File: backend/objectiv_backend/schema/event_schemas.py [hunk 1]
```diff
@@ -157,10 +157,12 @@ def get_event_schema(self, event_type: EventType) -> Optional[Dict[str, Any]]:
                     value['items']['type'] = 'object'
                 properties[key] = deepcopy(value)
 
+        # only add the property if it is required
```
[the remark this answers is not in the document]
Comment (borft) [reply backend/objectiv_backend/schema/event_schemas.py:160]: haha, sure, removed it ;)

**Comment by thijs-obj** [comment]
> > Changes look good to me, but do we not want to test the `validate_event_adheres_to_schema()` function too with some optional properties?
> 
> I think that is a good idea. However, as there are currently no optionals in the schema, I propose to add that as soon as there are (which will be when we add the `MarketingContext`)

sounds good to me

**Pull request merged**